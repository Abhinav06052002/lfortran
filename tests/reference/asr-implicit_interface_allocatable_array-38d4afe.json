{
    "basename": "asr-implicit_interface_allocatable_array-38d4afe",
    "cmd": "lfortran --show-asr --implicit-interface --no-color {infile} -o {outfile}",
    "infile": "tests/implicit_interface_allocatable_array.f90",
    "infile_hash": "6377d5c3ca768b2922097f929670903bdf047c2cfd355656dadfc6c5",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-implicit_interface_allocatable_array-38d4afe.stdout",
    "stdout_hash": "b027c7a57e27b59c653e46b501d1ebf91724564fc8e8f3b65923ebf3",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}