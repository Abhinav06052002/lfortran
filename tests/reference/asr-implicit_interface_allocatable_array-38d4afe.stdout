(TranslationUnit
    (SymbolTable
        1
        {
            implicit_interface_allocatable_array:
                (Function
                    (SymbolTable
                        2
                        {
                            r_g:
                                (Variable
                                    2
                                    r_g
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Allocatable
                                        (Array
                                            (Real 4)
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            sub:
                                (Function
                                    (SymbolTable
                                        3
                                        {
                                            sub_arg_0:
                                                (Variable
                                                    3
                                                    sub_arg_0
                                                    []
                                                    Unspecified
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    BindC
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            sub_arg_1:
                                                (Variable
                                                    3
                                                    sub_arg_1
                                                    []
                                                    Unspecified
                                                    ()
                                                    ()
                                                    Default
                                                    (Allocatable
                                                        (Array
                                                            (Real 4)
                                                            [(()
                                                            ())]
                                                            PointerToDataArray
                                                        )
                                                    )
                                                    ()
                                                    BindC
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    sub
                                    (FunctionType
                                        [(Integer 4)
                                        (Allocatable
                                            (Array
                                                (Real 4)
                                                [(()
                                                ())]
                                                PointerToDataArray
                                            )
                                        )]
                                        ()
                                        BindC
                                        Interface
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 3 sub_arg_0)
                                    (Var 3 sub_arg_1)]
                                    []
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    implicit_interface_allocatable_array
                    (FunctionType
                        []
                        ()
                        Source
                        Implementation
                        ()
                        .false.
                        .false.
                        .false.
                        .false.
                        .false.
                        []
                        .false.
                    )
                    []
                    []
                    [(SubroutineCall
                        2 sub
                        ()
                        [((IntegerConstant 1 (Integer 4) Decimal))
                        ((ArrayPhysicalCast
                            (Var 2 r_g)
                            DescriptorArray
                            PointerToDataArray
                            (Allocatable
                                (Array
                                    (Real 4)
                                    [(()
                                    ())]
                                    PointerToDataArray
                                )
                            )
                            ()
                        ))]
                        ()
                    )
                    (ImplicitDeallocate
                        [(Var 2 r_g)]
                    )]
                    ()
                    Public
                    .false.
                    .false.
                    ()
                )
        })
    []
)
